joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/astcreation/AstForExpressionsCreator.scala

```diff
@@ -37,11 +37,19 @@ trait AstForExpressionsCreator(implicit withSchemaValidation: ValidationMode) {
       case InterpolatedStringExpression      => astForInterpolatedStringExpression(expr)
       case ConditionalAccessExpression       => astForConditionalAccessExpression(expr)
       case SuppressNullableWarningExpression => astForSuppressNullableWarningExpression(expr)
+      case CoalesceExpression                => astForCoalesceExpression(expr)
       case _: BaseLambdaExpression           => astForSimpleLambdaExpression(expr)
       case _                                 => notHandledYet(expr)
     }
   }
 
+  private def astForCoalesceExpression(coalesceExpression: DotNetNodeInfo): Seq[Ast] = {
+    val leftAst  = astForExpression(createDotNetNodeInfo(coalesceExpression.json(ParserKeys.Left)))
+    val rightAst = astForExpression(createDotNetNodeInfo(coalesceExpression.json(ParserKeys.Right)))
+
+    leftAst ++ rightAst
```

Comment:
We should probably add a `<operator>.coalesce` so that we can embed the correct semantics

Reply:
Should this not also be a `callNode` or something similar with the operator you suggested above?

Reply:
@AndreiDreyer Yes, something like `callAst(callNode, leftAst ++ rightAst)`